### PR TITLE
[Backport 2.x] Add workflow to publish snapshots via GHA (#325)

### PR DESCRIPTION
**Pull request by VachaShah**
* Add workflow to publish snapshots via GHA

* Fix task name

* Fix conflict and add documentation

### Description
Backporting snapshot configuration from #325 

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,9 @@ This project has adopted the [Amazon Open Source Code of Conduct](CODE_OF_CONDUC
 
 See [User Guide](USER_GUIDE.md).
 
+## Snapshot Builds
+The [snapshots builds](https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/client/opensearch-java/) are published to sonatype using [publish-snapshots.yml](./.github/workflows/publish-snapshots.yml) workflow. Each `push` event to the main branch triggers this workflow.
```
Comment (gaiksaya) [inline README.md:56]: Triggers can change eventually so
```suggestion
The [snapshots builds](https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/client/opensearch-java/) are published to sonatype using [publish-snapshots.yml](./.github/workflows/publish-snapshots.yml) workflow.
```

**Review by gaiksaya — APPROVED** [review]
Minor nit. Looks good overall!

File: README.md (hunk 1 above)
Comment (VachaShah) [reply README.md]: Done.

**Comment by VachaShah** [comment]
Snapshot for the 2.x line is now available: https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/client/opensearch-java/2.5.0-SNAPSHOT/

**Pull request merged**